Support `dynamic`

Greg-Smulko opened the pull request:
Support `dynamic`.

Tested end-to-end:
![image](https://user-images.githubusercontent.com/26791344/201895479-644aa6f4-37ad-49b7-aca5-2d41ab6774a8.png)

Greg-Smulko committed:
Support dynamics

meziantou commented on Meziantou.FluentAssertionsAnalyzers.Tests/NUnitToFluentAssertionsAnalyzerUnitTests.cs:
```diff
@@ -51,7 +51,69 @@ public void MyTest()
+    public void MyTest()
+    {
+        dynamic val = new object();
+        val.Should().NotBeNull();
```
Are you sure this syntax is valid?

````
dynamic a = new object();
a.Should().BeNull();
````

> Unhandled exception. Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: 'object' does not contain a definition for 'Should'

Greg-Smulko replied:
Yes, that's a very fair point... 🤦 

As per https://stackoverflow.com/a/71676317/532575, I'm casting it to an `object` if needed.

It feels a bit ugly, and the resulting code is slightly more complicated too, so I won't be offended if you don't want to merge it at all. ;)

Greg-Smulko committed:
Extract Rewriter class

Greg-Smulko committed:
Make Rewriter non-static

Greg-Smulko committed:
Cast dynamic to object

To work around:
> Unhandled exception. Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: 'object' does not contain a definition for 'Should'
